pkg/3scale/amp/operator/apicast_options_provider.go

```diff
@@ -4,23 +4,31 @@ import (
 	"fmt"
 	"strconv"
 
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appsv1alpha1 "github.com/3scale/3scale-operator/apis/apps/v1alpha1"
 	"github.com/3scale/3scale-operator/pkg/3scale/amp/component"
 	"github.com/3scale/3scale-operator/pkg/3scale/amp/product"
 	"github.com/3scale/3scale-operator/pkg/helper"
-
-	appsv1alpha1 "github.com/3scale/3scale-operator/apis/apps/v1alpha1"
-	v1 "k8s.io/api/core/v1"
 )
 
 type ApicastOptionsProvider struct {
 	apimanager     *appsv1alpha1.APIManager
+	namespace      string
 	apicastOptions *component.ApicastOptions
+	client         client.Client
+	secretSource   *helper.SecretSource
 }
 
-func NewApicastOptionsProvider(apimanager *appsv1alpha1.APIManager) *ApicastOptionsProvider {
+func NewApicastOptionsProvider(apimanager *appsv1alpha1.APIManager, namespace string, client client.Client) *ApicastOptionsProvider {
```

Comment:
I think `namespace`  namespace might not be needed. We can use the `APIManager` namespace as I understand they will always be in the same namespace

Reply:
updated